test/functional/test_runner.py

```diff
@@ -322,6 +319,16 @@ def main():
         print("Rerun ./configure with --enable-wallet, --with-utils and --with-daemon and then make")
         sys.exit(0)
 
+    if not (len(ALL_SCRIPTS) == dict.fromkeys(ALL_SCRIPTS).keys()):
+        print("{}WARNING!{} There are duplicate tests in the test list:".format(BOLD[1], BOLD[0]))
+        duplicates = set()
+        for test in ALL_SCRIPTS:
```

Comment:
This algorithm seems inefficient (I know it's a test and who cares, but my perfectionism does care). You count these scripts twice: in `dict.fromkeys` and here. I suggest using multiset to count them once:

```python
from collections import Counter
unique_scripts = Counter(ALL_SCRIPTS)
if len(ALL_SCRIPTS) != len(unique_scripts):
    for s in unique_scripts:
        if unique_scripts.get(s) > 1:
            print('dup', s)
```